lib/datasource/artifactory/index.ts

```diff
@@ -0,0 +1,102 @@
+import { logger } from '../../logger';
+import { cache } from '../../util/cache/package/decorator';
+import { parse } from '../../util/html';
+import { HttpError } from '../../util/http/types';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import { datasource } from './common';
+
+export class ArtifactoryDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly defaultRegistryUrls = [];
+
```

Comment:
```suggestion
```

Remove, it's optional.

Reply:
I am confused here. I understood in a [previous comment](https://github.com/renovatebot/renovate/pull/11602#discussion_r703334308) that you suggested not removing it if there was no default registry.

In any case, I have removed the line in my last push as the tests are successful without it.